File: sem4/mpp/ex2/networking/src/main/java/networking/server/JsonConcurrentServer.java
package networking.server;
import networking.worker.AbstractWorker;
import networking.worker.MasterJsonWorker;
import services.IMasterService;

import java.net.Socket;

public class JsonConcurrentServer extends AbstractConcurrentServer {
	private final IMasterService service;

	public JsonConcurrentServer(int port, IMasterService service) {
		super(port);
		this.service = service;
		System.out.println("JsonConcurrentServer");
	}

	@Override
	protected Thread createWorker(Socket client) {
		AbstractWorker worker = new MasterJsonWorker(service, client);
		return new Thread(worker);
	}
}

File: sem4/mpp/project-cs/networking/Server/JsonConcurrentServer.cs
using System.Net.Sockets;
using networking.Worker;
using project_cs.utils;

namespace networking.Server;

public class JsonConcurrentServer : AbstractConcurrentServer {
    private IMasterService server;
    private MasterJsonWorker worker;

    public JsonConcurrentServer(string host, int port, IMasterService server) : base(host, port) {
        this.server = server;
        Console.WriteLine("SerialChatServer...");
    }

    protected override Thread CreateWorker(TcpClient client) {
        worker = new MasterJsonWorker(server, client);
        return new Thread(worker.run);
    }
}